lib/postrank-uri.rb

```diff
@@ -89,6 +89,7 @@ module URI
 
     URIREGEX[:escape]   = /([^ a-zA-Z0-9_.-]+)/x
     URIREGEX[:unescape] = /((?:%[0-9a-fA-F]{2})+)/x
+    URIREGEX[:reserved] = /(%(?:21|23|24|25|26|27|28|29|2A|2B|2C|2F|3A|3B|3D|3F|40|5B|5D))/x
```

Comment:
It is that list plus the percent, since once you decode the percent you cannot decode the string containing it again without potentially corrupting its contents.

That's a good thought about building this list from the list of characters.